train.py

```diff
@@ -312,47 +315,46 @@ def log_step(step_type, device, step, log_output=None, tracker=None):
             )
         return msg
 
-    def prepare_task(args, devices):
-        # Setup task, e.g., translation, language modeling, etc.
-        task = tasks.setup_task(args)
-
-        # Load valid dataset (we load training data below, based on the latest checkpoint)
-        for valid_sub_split in args.valid_subset.split(','):
-            task.load_dataset(valid_sub_split, combine=True, epoch=0)
-
-        # Build models and criteria to print some metadata
-        model_parallel = dp.DataParallel(
-            lambda: task.build_model(args), device_ids=devices)
-        model, criterion = task.build_model(args), task.build_criterion(args)
-        print_model_criterion(model, criterion, args)
-        del model, criterion
-
-        # Build trainers
-        trainers = {
-            device: Trainer(args, task, model, task.build_criterion(args), xla=True)
-            for device, model in zip(model_parallel.devices, model_parallel.models)
-        }
-        lr = trainers[devices[0]].get_lr()
-
-        # Load the latest checkpoint if one is available and restore the
-        # corresponding train iterator
-        extra_state, epoch_itr = checkpoint_utils.load_checkpoint(
-            args, trainers[devices[0]])
-        if extra_state is not None:
-            # checkpoint detected, load saved model weights to all devices
-            xu.eprint(
-                'checkpoint detected, device 0 meters need to be '
-                're-loaded to device'
-            )
-            checkpoint_utils.load_checkpoint_tpu(args, trainers, devices[0])
-        valid_subsets = args.valid_subset.split(',')
-        return task, trainers, model_parallel, epoch_itr, lr, valid_subsets
-
-    def train_loop_fn(model, loader, device, context):
-        trainer = trainers[str(device)]
-        stats, log_output = None, None
-        tracker = xm.RateTracker()
+    def prepare_task(args, xla_device):
+       # Setup task, e.g., translation, language modeling, etc.
```

Comment:
is indenting OK here? Looks like 3 spaces in Github UI

Reply:
good catch, thanks